keccak256/src/arith_helpers.rs

```diff
@@ -184,3 +201,54 @@ pub fn inspect(x: BigUint, name: &str, base: u64) {
     }
     println!("inspect {} {} info {:?}", name, x, info);
 }
+
+pub fn state_to_biguint<F: FieldExt>(state: [F; 25]) -> StateBigInt {
+    StateBigInt {
+        xy: state
+            .iter()
+            .map(|elem| elem.to_bytes())
+            .map(|bytes| BigUint::from_bytes_le(&bytes))
+            .collect(),
+    }
+}
+
+pub fn state_to_state_bigint<F: FieldExt>(state: [F; 17]) -> State {
+    let mut matrix = [[0u64; 5]; 5];
+
+    let mut elems: Vec<u64> = state
+        .iter()
+        .map(|elem| elem.to_bytes())
+        // This is horrible. But FieldExt does not give much better alternatives and
+        // refactoring `State` will be done once the keccak_all_togheter is done.
+        .map(|bytes| {
+            let mut arr = [0u8; 8];
+            arr.copy_from_slice(&bytes[0..8]);
+            u64::from_le_bytes(arr)
+        })
+        .collect();
+    elems.extend(vec![0u64; 25 - 17]);
+    (0..5)
+        .into_iter()
+        .for_each(|idx| matrix[idx].copy_from_slice(&elems[0 * idx..5 * idx]));
```

Comment:
Is this range `[0 * idx..5 * idx]` what we want? As `idx` goes from `0..5`, we get the ranges `[0..0], [0..5], [0..10], [0..15], [0..20]`.

Reply:
Ohhh. So the first one is the wrong one and instead we lack the 25 one.

With `(1..6)` I think we will get it right since that should produce the ranges: `[0..5], [5..10], [10..15], [15..20], [20..25]`.

It's a bit annoying how `State` is constructed/that is required for `next_inputs` in that shape.

Reply:
Should the subset range be: `[(5 * idx)..(5 * idx + 5)]`? For `idx` from `0..5`, I think that gives us the desired ranges.

Reply:
Isn't it what I have above?? Maybe I'm missunderstanding your message

Reply:
Yes, I think I had seen a version of the comment before it was edited.